build-tools/packages/build-cli/src/library/repoPolicyCheck/npmPackages.ts

```diff
@@ -1973,6 +1974,44 @@ export const handlers: Handler[] = [
 			return result;
 		},
 	},
+	{
+		name: "npm-check-unused-dependencies",
+		match,
+		handler: async (file: string): Promise<string | undefined> => {
+			const depcheckConfigFileName = ".depcheckrc.cjs";
+			const packageDir = path.resolve(path.dirname(file));
+			const depcheckConfigFilePath = path.resolve(
+				path.join(packageDir, depcheckConfigFileName),
+			);
```

Comment:
I am assuming that every package in our repo can add optional customization to the depcheck tool, if they wish to. So the best placed seemed to be right next to the package.json file, just like eslintrc. Note that providing the config file is optional, it is required only in case of customizations.

If there is someway that depcheck command can be run from the CLI as a part of these repo-checks, then depcheck will automatically look for a file called `.depcheckrc`. Since we are using the API route here instead, depcheck api expects the `options` to be provided into the function. Hence a file read for each package is required.

cc @sonalideshpandemsft  / @jason-ha

Reply:
If depcheck were run from command line, would it look up folder hierarchies to find .depcheckrc?
What is its exact search behavior? (Could it search in current directory or would it only search next to a package.json file?)